modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/TestcontainersExtension.java

```diff
@@ -0,0 +1,125 @@
+package org.testcontainers.junit.jupiter;
+
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
+import org.junit.jupiter.api.extension.TestInstancePostProcessor;
+import org.junit.platform.commons.support.AnnotationSupport;
+import org.junit.platform.commons.util.Preconditions;
+import org.junit.platform.commons.util.ReflectionUtils;
+import org.testcontainers.lifecycle.Startable;
+
+import java.lang.reflect.Field;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+class TestcontainersExtension implements TestInstancePostProcessor, BeforeEachCallback {
+
+    private static final Namespace NAMESPACE = Namespace.create(TestcontainersExtension.class);
+
+    private static final String TEST_INSTANCE = "testInstance";
+
+    @Override
+    public void postProcessTestInstance(final Object testInstance, final ExtensionContext context) {
+        ExtensionContext.Store store = context.getStore(NAMESPACE);
+        store.put(TEST_INSTANCE, testInstance);
+
+        findSharedContainers(testInstance)
+            .forEach(adapter -> store.getOrComputeIfAbsent(adapter.key, k -> adapter.start()));
```

Comment:
Any reason to use field instead of getter?

Reply:
The `StoreAdapter` class is a private class inside of `TestcontainersExtension`. So all fields are visible from `TestcontainersExtension` and there is no gain from an information hiding perspective if we add getters.

Reply:
The problem I see is that it is inconsistent with the rest of our code base because we always use getters :)